pallets/gear/src/tests.rs

```diff
@@ -4136,6 +4308,10 @@ mod utils {
             .expect("can't find message send event")
     }
 
+    pub(super) fn maybe_last_mail(account: AccountId) -> Option<StoredMessage> {
+        MailboxOf::<Test>::iter_key(account).last()
```

Comment:
order isn't guaranteed in mailbox
so the best solution to get last mail is to iter events and find the latest UserMessageSent to given account_id